lib/ParseSIL/ParseSIL.cpp

```diff
@@ -2986,29 +2956,24 @@ bool SILParser::parseSILInstruction(SILBuilder &B) {
                  diag::sil_inst_autodiff_expected_function_type_operand);
       return true;
     }
-    SmallVector<SILValue, 16> associatedFunctions;
-    // Parse optional operand lists `with { <operand> , <operand> }, ...`.
+    Optional<std::pair<SILValue, SILValue>> derivativeFunctions = None;
+    // Parse an optional operand list `with { <operand> , <operand> }`.
     if (P.Tok.is(tok::identifier) && P.Tok.getText() == "with") {
       P.consumeToken(tok::identifier);
-      // Parse associated function values as operand lists. There are as many
-      // operand lists as the differentiation order.
-      associatedFunctions.reserve(2 * order);
-      for (unsigned listIdx = 0; listIdx < order; ++listIdx) {
-        // FIXME(rxwei): Change this to *not* require a type signature once
-        // we can infer AD associated function types.
-        SILValue newAssocFn1, newAssocFn2;
-        if (P.parseToken(tok::l_brace,
-                diag::sil_inst_autodiff_operand_list_expected_lbrace) ||
-            parseTypedValueRef(newAssocFn1, B) ||
-            P.parseToken(tok::comma,
-                diag::sil_inst_autodiff_operand_list_expected_comma) ||
-            parseTypedValueRef(newAssocFn2, B) ||
-            P.parseToken(tok::r_brace,
-                         diag::sil_inst_autodiff_operand_list_expected_rbrace))
-          return true;
-        associatedFunctions.push_back(newAssocFn1);
-        associatedFunctions.push_back(newAssocFn2);
-      }
+      // Parse associated function values as an operand list.
+      // FIXME(rxwei): Change this to *not* require a type signature once
```

Comment:
`FIXME` follow-up: is it desirable to "not require type signatures for `{ %jvp, %vjp }` operands"?

Reply:
Yes, because operands' types can be inferred directly from the original function. Similarly, `apply`'s arguments do not have type signatures printed.